cryostatio/cryostat
# feat(uploads): add maxFiles configuration policy to jvmId uploaded recordings

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #1306

## Description of the change:
This change allows any client which posts a recording to the target-specific upload handler `RecordingsFromIdPostHandler` to specify the maxFiles parameter which keeps a set amount of archived recording files per target, depending on the given parameter. If the parameter is not specified, then it is defaulted to the env variable `CRYOSTAT_PUSH_MAX_FILES`. If that env var is not specified, the global maxFiles value is `Integer.MAX_INT`.

## Motivation for the change:
See #1306 for discussion.

## How to manually test:
1. Run normally...
2. On a target e.g. (cryostat:9093), start a recording and archive it (turn off archiveOnStop to make it easier).
3. Find the target's `jvmId`.
4. Then ->
```bash
$ https --verify=no -f -v POST :8181/api/beta/recordings/${jvmId} recording@testfolder/jfr/quarkus-test-agent_default_20230109T001426Z.jfr maxFiles=1 --auth=user:pass 
```
Notice the extra new paramter `maxFiles`. If we set it to 1, it will delete all archived files within the target except the one that was included in this upload form. If set to anything above 1, it will keep the old archived recording, plus the recent uploaded one that is included in the form.

Experiment with settings, (e.g. without a `maxFiles`, this should default to the environment variable config that was either set, or defaulted to `Integer.MAX_INT`)
